plugins/hypervisors/vmware/src/main/java/com/cloud/storage/resource/VmwareStorageProcessor.java

```diff
@@ -2025,7 +2025,6 @@ public Answer backupSnapshot(CopyCommand cmd) {
                 try {
                     if (workerVm != null) {
                         // detach volume and destroy worker vm
-                        workerVm.detachAllDisks();
```

Comment:
@harikrishna-patnala can you check if per the new code/manager, we should detach the worker VM's disks after the backup of snapshot to secondary storage passes?
cc @nvazquez @sureshanaparti

Reply:
verified the issue and is reproducible. We need to delete the cloned VM completely including disks, not the worker VM (worker VM has the actual disk attached to it). Also there is another place to fix. I'll fix and update this PR. Thanks @rhtyd for drafting this PR.